libs/cart/src/drivers/interfaces/cart-coupon-service.interface.ts

```diff
@@ -0,0 +1,19 @@
+import { InjectionToken } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { DaffCartCoupon } from '../../models/cart-coupon';
+import { DaffCart } from '../../models/cart';
+
+/**
+ * The interface responsible for apply a coupon to a cart.
+ */
+export interface DaffCartCouponServiceInterface<T extends DaffCart = DaffCart, V extends DaffCartCoupon = DaffCartCoupon> {
+	/**
+	 * Apply a coupon to the cart and return a partial of the cart.
+	 */
+  applyCoupon(cartId: T['id'], coupon: V): Observable<Partial<T>>;
```

Comment:
Should we be able to add/remove/list all coupons on a cart out of the gate, or should we wait for another PR?

Reply:
I always thought of coupons as coupon codes that you only know about if you've been sent an email, or you follow someone on social media who advertises a product and provides a coupon code. Being able to add/list coupons from outside the admin panel doesn't make sense to me. Removing coupons I could see as useful, though I doubt anyone would really use it. I'll look into removing coupons

Reply:
Lost Golf Balls actively advertises its coupon codes everywhere on the website. There is a section on the product page that adds relevant coupon codes to the product, so I do think it makes sense to have a section to list coupons available.

i.e. Sephora has a page that lists all of their available coupon codes, and it is referenced as a link on the cart page.

I think this is a separate PR though.